fixieai/agents-ts/src/serve.ts

```diff
@@ -0,0 +1,213 @@
+ *    2. From ts-node (This is for local dev.)
+ *
+ * In both cases, we want to be able to `require` an Agent written in TS.
+ * In case (1), we need to call tsNode.register() to enable that.
+ * In case (2), we don't need to call tsNode.register(), because we're already in ts-node. And if we do call it, it
+ * actually creates problems.
+ */
+// @ts-expect-error
+if (!process[Symbol.for('ts-node.register.instance')]) {
+  tsNode.register();
+}
+
+/**
+ * This should be kept in sync with the `AgentConfig` dataclass.
+ */
+export interface AgentConfig {
+  handle: string;
+  name?: string;
+  description: string;
+  more_info_url: string;
+  language: string;
+  entry_point: string;
+  deployment_url?: string;
+  public?: boolean;
+}
+
+interface AgentMetadata {
+  base_prompt: string;
+  few_shots: string[];
+}
+
+interface AgentResponse {
+  message: Message;
+}
+interface Message {
+  text: string;
+}
+
+type AgentFunc = (args: string) => string;
+
+interface Agent {
+  basePrompt: string;
+  fewShots: string[];
+  funcs: Record<string, AgentFunc>;
+}
+
+class FunctionNotFoundError extends Error {
+  name = 'FunctionNotFoundError';
+}
+
+class AgentRunner {
+  private readonly agent: Agent;
+
+  constructor(agentPath: string) {
+    const requiredAgent = require(agentPath);
+    const allExports = Object.keys(requiredAgent).join(', ');
+
+    if (typeof requiredAgent.BASE_PROMPT !== 'string') {
+      throw new Error(
+        `Agent must have a string export named BASE_PROMPT. The agent exported the following: ${allExports}.`,
+      );
+    }
+    if (typeof requiredAgent.FEW_SHOTS !== 'string') {
+      throw new Error(
+        `Agent must have a string export named FEW_SHOTS. The agent exported the following: ${allExports}.`,
+      );
+    }
+    const funcs = _.omit(requiredAgent, 'BASE_PROMPT', 'FEW_SHOTS');
+
+    this.agent = {
+      basePrompt: requiredAgent.BASE_PROMPT,
+      fewShots: requiredAgent.FEW_SHOTS.split('\n\n'),
+      funcs,
+    };
+  }
+
+  runFunction(funcName: string, args: Parameters<AgentFunc>[0]): ReturnType<AgentFunc> {
+    const func = this.agent.funcs[funcName];
+    if (!func) {
+      throw new FunctionNotFoundError(
+        `Function not found: ${funcName}. Functions available: ${Object.keys(this.agent.funcs).join(', ')}`,
+      );
+    }
+    return func(args);
+  }
+
+  getAgentMetadata(): AgentMetadata {
+    return {
+      base_prompt: this.agent.basePrompt,
+      few_shots: this.agent.fewShots,
+    };
+  }
+}
+
+/**
+ * TODO:
+ *  - watch mode
+ *  - logger formatting for local dev
+ *  - async agent functions
+ */
+
+export default async function serve({
+  agentConfigPath,
+  agentConfig,
+  port,
+  silentStartup,
+  refreshMetadataAPIUrl,
+  silentRequestHandling = false,
+  humanReadableLogs = false,
+}: {
+  agentConfigPath: string;
+  agentConfig: AgentConfig;
+  port: number;
+  silentStartup: boolean;
+  refreshMetadataAPIUrl: string;
```

Comment:
Assuming this code will eventually be used in the cloud as well, I think this should be optional (we'll want `fixie serve` to use it on changes but not `fixie deploy` on startup).